helpers/test-data-generator.js

```diff
@@ -15,17 +15,25 @@ import {
   RATING_2_SALT,
   MANAGER_ROLE,
   EVALUATOR_ROLE,
-  WORKER_ROLE,
-  SPECIFICATION_HASH
+  WORKER_ROLE
 } from "../helpers/constants";
 import { currentBlockTime, createSignatures } from "../helpers/test-helper";
 
 const IColony = artifacts.require("IColony");
 const Token = artifacts.require("Token");
 
 export async function setupAssignedTask({ colonyNetwork, colony, dueDate, domain = 1, skill = 0, evaluator = EVALUATOR, worker = WORKER }) {
-  await colony.makeTask(SPECIFICATION_HASH, domain);
+  // By using a random specification hash rather than SPECIFICATION_HASH, we can create tasks in parallel and pick up
+  // the correct taskID afterwards.
+  const specificationHash = web3Utils.randomHex(32);
   let taskId = await colony.getTaskCount.call();
+  taskId = taskId.minus(1);
```

Comment:
Remember that somehow magically our task ids have become 1-based ..

Reply:
Or is this because you are adding `1` to the `id` at the start of the loop after?

Reply:
No, this was a mistake. Doesn't matter now though, because of the other, better way of using events you suggested 😄